sample/normal_double.go

```diff
@@ -0,0 +1,133 @@
+/*
+* Copyright (c) 2018 XLAB d.o.o
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package sample
+
+import (
+	"math/big"
+	"crypto/rand"
+	"fmt"
+)
+
+// NormalDouble samples random values from the
+// normal (Gaussian) probability distribution, centered on 0.
+// This sampler works in a way that first samples form a
+// NormalCumulative with some small sigma and then using
+// another sampling from uniform distribution creates a candidate
+// for the output, which is accepted or rejected with certain
+// probability.
+type NormalDouble struct {
+	*Normal
+	// NormalCumulative sampler used in the first part
+	samplerCumu *NormalCumulative
+	// precomputed parameters used for sampling
+	k *big.Int
+	twiceK *big.Int
```

Comment:
Stick to one naming scheme - here, `2 * k` is named `twiceK`, but in `NormalNegative` sampler, `2 * cut + 1` is named `cutTimes2`.